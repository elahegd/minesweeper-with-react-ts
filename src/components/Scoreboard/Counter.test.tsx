import React from 'react';
import { render, screen, fireEvent, cleanup } from "@testing-library/react";

import {Counter} from "./Counter";


afterEach(cleanup);

describe("Test click counter", () => {
    it("should increse the counter when click on the button", () => {
        const { asFragment } = render(<Counter>001</Counter>);

        expect(asFragment()).toMatchSnapshot();
        // const counter = screen.getByRole("heading");
        // const increaseButton = screen.getByRole("button", {name : "Increse"});
        // const decreaseButton = screen.getByRole("button", {name: "Decrease"});

        // expect(counter.textContent).toBe("Counter: 0");

        // fireEvent.click(increaseButton);
        // expect(counter.textContent).toBe("Counter: 1");

        // fireEvent.click(decreaseButton);
        // expect(counter.textContent).toBe("Counter: 0");
    })
})