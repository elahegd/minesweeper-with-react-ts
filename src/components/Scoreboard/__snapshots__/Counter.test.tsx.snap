// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test click counter should increse the counter when click on the button 1`] = `
<DocumentFragment>
  <div
    class="css-d2q1pa"
  >
    001
  </div>
</DocumentFragment>
`;
